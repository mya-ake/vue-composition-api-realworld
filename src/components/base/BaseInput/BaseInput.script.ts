import { createComponent } from '@vue/composition-api';
import { useModel } from '@/compositions/form';
import { BaseInputType } from './BaseInput.type';

export default createComponent({
  props: {
    type: { default: 'text' },
    value: { required: true },
  },
  setup(props: BaseInputType.Props, { emit }) {
    const { model } = useModel(props, emit);
    return { model };
  },
});
