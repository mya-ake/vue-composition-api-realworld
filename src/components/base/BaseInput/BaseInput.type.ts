export namespace BaseInputType {
  export interface Props {
    type?: string;
    value: string;
  }
}
