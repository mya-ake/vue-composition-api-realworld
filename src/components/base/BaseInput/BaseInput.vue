<template>
  <input v-model="model" :type="type" class="form-control" />
</template>

<script lang="ts" src="./BaseInput.script.ts"></script>
