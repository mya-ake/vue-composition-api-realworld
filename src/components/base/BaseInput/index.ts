export { default as BaseInput } from './BaseInput.vue';
export * from './BaseInput.type';
