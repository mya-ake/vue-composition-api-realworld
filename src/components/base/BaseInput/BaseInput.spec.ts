import { shallowMount, Wrapper } from '@vue/test-utils';
import BaseInput from './BaseInput.vue';

describe('components/base/BaseInput', () => {
  describe('mountable', () => {
    let wrapper: Wrapper<Vue>;

    beforeEach(() => {
      wrapper = shallowMount(BaseInput, { propsData: { value: '' } });
    });

    it('is vue instance', () => {
      expect(wrapper.isVueInstance()).toBe(true);
    });

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
