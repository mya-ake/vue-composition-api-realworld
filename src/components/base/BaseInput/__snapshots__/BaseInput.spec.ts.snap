// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/contents/BaseInput mountable snapshot 1`] = `
<input
  class="form-control"
  type="text"
/>
`;
